packages/xrpl/test/integration/submitAndWait.test.ts

```diff
@@ -1,91 +1,151 @@
-        assert.equal(response.result.validated, true)
-      },
-    )
-  })
+  let testContext: XrplIntegrationTestContext
 
-  it('should throw a ValidationError when submitting an unsigned transaction without a wallet', async function () {
-    const accountSet: AccountSet = {
-      TransactionType: 'AccountSet',
-      Account: this.wallet.classicAddress,
-      Domain: convertStringToHex('example.com'),
-    }
-
-    await assertRejects(
-      this.client.submitAndWait(accountSet),
-      ValidationError,
-      'Wallet must be provided when submitting an unsigned transaction',
-    )
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
   })
+  afterEach(async () => teardownClient(testContext))
 
-  it('submitAndWait a signed transaction', async function () {
-    const accountSet: AccountSet = {
-      TransactionType: 'AccountSet',
-      Account: this.wallet.classicAddress,
-      Domain: convertStringToHex('example.com'),
-    }
-    const { tx_blob: signedAccountSet } = this.wallet.sign(
-      await this.client.autofill(accountSet),
-    )
-    const responsePromise = this.client.submitAndWait(signedAccountSet)
-    const ledgerPromise = setTimeout(ledgerAccept, 1000, this.client)
-    return Promise.all([responsePromise, ledgerPromise]).then(
-      ([response, _ledger]) => {
-        assert.equal(response.type, 'response')
-        assert.equal(response.result.validated, true)
-      },
-    )
-  })
+  async function delayedLedgerAccept(
+    context: XrplIntegrationTestContext,
+  ): Promise<unknown> {
+    await new Promise<void>((resolve) => {
+      setTimeout(resolve, 1000)
+    })
+    return ledgerAccept(context.client)
+  }
 
-  it('submitAndWait a signed transaction longer', async function () {
-    const accountSet: AccountSet = {
-      TransactionType: 'AccountSet',
-      Account: this.wallet.classicAddress,
-      Domain: convertStringToHex('example.com'),
-    }
-    const { tx_blob: signedAccountSet } = this.wallet.sign(
-      await this.client.autofill(accountSet),
-    )
-    const responsePromise = this.client.submitAndWait(signedAccountSet)
-    const ledgerPromise = setTimeout(ledgerAccept, 5000, this.client)
-    return Promise.all([responsePromise, ledgerPromise]).then(
-      ([response, _ledger]) => {
-        assert.equal(response.type, 'response')
-        assert.equal(response.result.validated, true)
-      },
-    )
-  })
+  it(
+    'submitAndWait an unsigned transaction',
+    async () => {
+      const accountSet: AccountSet = {
+        TransactionType: 'AccountSet',
+        Account: testContext.wallet.classicAddress,
+        Domain: convertStringToHex('example.com'),
+      }
+
+      let retries = 10
+
+      while (retries > 0) {
+        retries -= 1
+        const responsePromise = testContext.client.submitAndWait(accountSet, {
+          wallet: testContext.wallet,
+        })
+        const ledgerPromise = delayedLedgerAccept(testContext)
+
+        try {
```

Comment:
Isn't this code already in a helper method somewhere?

Reply:
No, because the code is slightly different than the other places.